tmt-app/src/Components/Breadcrumb.jsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { Breadcrumbs, Typography } from "@mui/material";
+import ForwardArrow from "../Images/Forward.png";
+import { Link } from "react-router-dom";
+
+function Breadcrumb({ text, page, previousPage }) {
+  return (
+    <Breadcrumbs
+      separator={
+        <img
+          style={{ width: "24px", margin: "0" }}
+          src={ForwardArrow}
+          alt="ForwardArrow"
+        />
+      }
+      aria-label="breadcrumb"
+      sx={{
+        display: "flex",
+        alignItems: "center",
+        textAlign: "center",
+        marginX: {
+          xs: "1rem",
+          md: "4rem",
+        },
+      }}
+    >
+      <Link to={`/${previousPage}`} style={{ textDecoration: "none" }}>
+        <Typography
+          variant="h6"
+          sx={{
+            color: "BobaHighlight.main",
+            marginRight: {
+              xs: "0px",
+              sm: "8px",
+              md: "16px",
+            },
+          }}
+        >
+          All Careers & Majors
+        </Typography>
+      </Link>
+      <Typography
+        variant="h6"
+        sx={{
+          color: "BobaHighlight.main",
+          marginLeft: {
+            xs: "0px",
+            sm: "8px",
+            md: "16px",
+          },
+        }}
+      >
+        {text} ({page})
+      </Typography>
```

Comment:
https://github.com/BobaTalks/pick-a-major/pull/106#issuecomment-1537815746

Pretty much asking you here to see if we can think a little bigger picture. We can still consider a more dynamic and scalable approach to using breadcrumbs that isn't tied only to this.

While a limited # of route possibilities might make this "work" i feel we could still keep some higher standards for ourselves and find something a little more clever that might be a pattern you can reuse in the future in work as well

Trying to shift the thought towards the concept of reusability in our code

we have access to `props` like you mentioned and this is a core feature of react we want to leverage. but if `props.previousPage` changes from a `string` type to `string[]` type, or even `object[]` there are some new possibilities that open up for us here.

if we think of this with a little bit more `type` strictness we can look at what a `Route` type might be. we know that we need to provide a `label` for the user to easily read, and a `url` or `path` that the breadcrumb actually links to. At a bare minimum these two properties can be the make up of a `Route`.

```js
const breadcrumbs = [
  { url: '/', label: 'Home' },
  { label: `Major: ${props.pageTitle}` }
]
```
With something like the structure above, we're essentially creating breadcrumbs as an iterabable array that gives us more power and flexibility within the component. so we can say in our mind that `breadcrumbs` is an array of `Breadcrumb`s where `Breadcrumb` is going to be `{ label: string, url?: string }` with an optional url (to see in a second)

```jsx
<Breadcrumbs>
  {breadcrumbs.map(breadcrumb => {
    return breadcrumb?.url 
      ? <Link to={breadcrumb.url}>{breadcrumb.label}</Link>
      : <Typography>{breadcrumb.label}</Typography>
    }
  )}
</Breadcrumbs>
```

With an implementation like this, we're now able to drop this breadcrumb into any page, and pass in an array of "breadcrumbs" and create that structure with any new and existing pages

Hope this abstraction makes sense, it applies to this use case, but it also applies to so many others, hoping to direct you to start to identify some of those opportunities in other areas of the app, and your work outside of BT too.

Some reading material:
https://bradfrost.com/blog/post/atomic-web-design/
https://react.dev/learn/thinking-in-react

Reply:
This is really neat! I never really thought about it this way. I will read more on this topic and experiment more on it.